### DISCUSSION TITLE
iOS crash while drawing anything

### DISCUSSION THREAD

**Discussion by PaulMcGinley** [MonoGame/MonoGame#8438] [iOS / iPadOS]
While on my M3 MacBook, using .net8 I get this error while trying to draw anything:

`/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8020/tools/bin/mlaunch --launchsim /Users/paul/Desktop/mg_ios_test/src/iosGame/bin/Debug/net8.0-ios/iossimulator-arm64/iosGame.app --device=:v2:udid=6527E5B3-389D-43DF-878F-4EA049388885 -sdk 17.4 --argument=-monodevelop-port --argument=10001 --setenv=__XAMARIN_DEBUG_PORT__=10001 --sdkroot /Applications/Xcode.app/Contents/Developer
xcrun simctl launch --console --terminate-running-process 6527E5B3-389D-43DF-878F-4EA049388885 project.MonoGame.iosGame -monodevelop-port 10001
project.MonoGame.iosGame: 20804
2024-08-14 16:10:39.582157+0100 iosGame[20804:3252724] Unbalanced calls to begin/end appearance transitions for <Microsoft_Xna_Framework_iOSGameViewController: 0x1147103c0>.
2024-08-14 16:10:39.599801+0100 iosGame[20804:3252724] void * _Nullable NSMapGet(NSMapTable * _Nonnull, const void * _Nullable): map table argument is NULL

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGBUS while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x113d281dc - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_dump_native_crash_info
	0x113cd83c8 - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_handle_native_crash
	0x113c3f914 - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_sigsegv_signal_handler_debug
	0x1120a37e0 - /usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x1b8650ff8 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCoreVMClient.dylib : cvmsServerElementBuild
	0x1b864ce34 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCoreVMClient.dylib : cvms_element_build_from_source
	0x1b8643854 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCVMSPluginSupport.dylib : _Z26cvm_deferred_build_modularPv
	0x180171978 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_client_callout
	0x180179b10 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_lane_serial_drain
	0x18017a688 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_lane_invoke
	0x180185a84 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_root_queue_drain_deferred_wlh
	0x1801850d0 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_workloop_worker_thread
	0x11358b814 - /usr/lib/system/libsystem_pthread.dylib : _pthread_wqthread
	0x11358a5d4 - /usr/lib/system/libsystem_pthread.dylib : start_wqthread

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x1120a3410):0x1120a3400  28 24 40 a8 2a 2c 41 a8 21 80 00 91 42 00 05 cb  ($@.*,A.!...B...
0x1120a3410  0c 34 00 a9 0e 3c 01 a9 42 00 01 f1 29 01 00 54  .4...<..B...)..T
0x1120a3420  68 24 00 a9 6a 2c 01 a9 63 80 00 91 28 24 40 a8  h$..j,..c...($@.
0x1120a3430  2a 2c 41 a8 21 80 00 91 42 80 00 f1 28 ff ff 54  *,A.!...B...(..T

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGBUS while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

An error has occurred in the native fault reporting. Some diagnostic information will be unavailable.

=================================================================
	Native stacktrace:
=================================================================
	0x113d281dc - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_dump_native_crash_info
	0x113cd83c8 - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_handle_native_crash
	0x113c3f914 - /Users/paul/Library/Developer/CoreSimulator/Devices/6527E5B3-389D-43DF-878F-4EA049388885/data/Containers/Bundle/Application/50F8BB78-B0D3-490A-A36E-7C8301A7D49C/iosGame.app/libmonosgen-2.0.dylib : mono_sigsegv_signal_handler_debug
	0x1120a37e0 - /usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x1b8650ff8 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCoreVMClient.dylib : cvmsServerElementBuild
	0x1b864ce34 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCoreVMClient.dylib : cvms_element_build_from_source
	0x1b8643854 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/OpenGLES.framework/libCVMSPluginSupport.dylib : _Z26cvm_deferred_build_modularPv
	0x180171978 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_client_callout
	0x180179b10 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_lane_serial_drain
	0x18017a688 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_lane_invoke
	0x180185a84 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_root_queue_drain_deferred_wlh
	0x1801850d0 - /Library/Developer/CoreSimulator/Volumes/iOS_21E213/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 17.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdispatch.dylib : _dispatch_workloop_worker_thread
	0x11358b814 - /usr/lib/system/libsystem_pthread.dylib : _pthread_wqthread
	0x11358a5d4 - /usr/lib/system/libsystem_pthread.dylib : start_wqthread

Exiting early due to double fault.
`

Using:
Rider
MonoGame 3.8.1.303
.Net8

Without the draw call the app starts up.

**Comment by stromkos** [comment]
I see that you are emulating the IOS on a Mac. 

Does the game crash when run on an actual IOS device? Or when run natively on the Mac with DesktopGL?

The emulator can hide run-time code errors.
